### PR TITLE
Forms: introduce RadioButtonGroup

### PR DESCRIPTION
**Pull request by dprokop**
Ref #19855 

Introduces `RadioButtonGroup` component for handling single select from multiple options.

@jessover9000 I have introduced change to how the focus is handled. According to the desicn the focus ring should be applied to the entire component:

image

Which I think does not make much sense, as user never knows where the focus is. Instead of that, the focus ring is applied to a single option:

image

@torkelo I have took the API approach similar to select where you pass options (`SelectableValue[]`) to the `RadioButtonGroup` and then it's this component that takes care for buttons rendering. This is different compared to `ToggleButtonGroup` which expects options to be rendered as children. Do you have any thoughts about this approach?

### CODE REVIEW

**Comment by torkelo** [comment]
Looks good! 

> I have introduced change to how the focus is handled. According to the desicn the focus ring should be applied to the entire component:

I think I discussed this with Patrick and we decided to not focus whole component and only single buttons.  So what you did was good :)

File: packages/grafana-data/src/types/theme.ts [hunk 1]
```diff
@@ -241,6 +241,15 @@ export interface GrafanaTheme extends GrafanaThemeCommons {
...
     formCheckboxCheckmark: string;
+    formRadioButtonText: string;
+    formRadioButtonTextActive: string;
+    formRadioButtonTextHover: string;
+    formRadioButtonBg: string;
+    formRadioButtonBgDisabled: string;
+    formRadioButtonBgActive: string;
+    formRadioButtonBorder: string;
+    formRadioButtonBorderHover: string;
+    formRadioButtonBorderActive: string;
```
Comment (torkelo) [inline packages/grafana-data/src/types/theme.ts]: supprised that we need so many new theme variables here. Is there a way to reduce these, and create more high level form color variables (border, highlight, hover etc). 

the alternative is to create local variables for these inside getRadioButtonStyles, not sure adding so many component-specific variables to the theme is the way to go. I remember we discussed this earlier in the year. Think some high-level form variables that multiple components can re-use is a good idea, but this feels a bit too much and component specific. 

thoughts?

Comment (dprokop) [reply packages/grafana-data/src/types/theme.ts]: Wasn't too specific about this, but majority of these variables are to be moved to the corresponding components.

Comment (torkelo) [reply packages/grafana-data/src/types/theme.ts]: and what's the reason for adding them here now?

Comment (dprokop) [reply packages/grafana-data/src/types/theme.ts]: No reason, just wanted to move all the variables at once. But yeah, makes sense to do it step by step.

Comment (dprokop) [reply packages/grafana-data/src/types/theme.ts]: @torkelo moved radio button vars to the component and removed from the theme. Will do the same for other components too. Until there's a need for exposing them for external usage I'll keep them private.

**Review by torkelo — APPROVED** [review]
Looks great! Excited to start using this in visualization settings instead of dropdowns with 2-3 options

**Pull request merged**